docs/syntax_and_semantics/while.md

```diff
@@ -46,3 +44,78 @@ loop do
   break if some_condition
 end
 ```
+
+## As an expression
+
+The value of a `while` is the value of the `break` expression that exits the `while`'s body:
+
+```crystal
+a = 0
+x = while a < 5
+  a += 1
+  break "four" if a == 4
+  break "three" if a == 3
+end
+x # => "three"
```

Comment:
This is still `nil` on the latest version of Crystal, so this PR must not be merged yet.

And I think we'll need to define some kind of policy for that:

1. When do such doc updates get merged?
    1. Should they be merged into a branch first?
2. How to indicate that a feature is new since some version?

Reply:
Good questions! But I think they should be asked in a dedicated issue, not in the comment thread on an affected pull request. 🙇‍♂️